### PR TITLE
Closes #1231: Migrate front page

### LINKED ISSUE #1231: Home page is broken on some migrated sites.
## Problem/Motivation
The Home page displays the error, 'Page not found'.

### Describe the bug

Home Page/Link:
•	The home link is not working.
•	The title is in Content, but it cannot be viewed or edited.
•	Page not found error.

### To Reproduce

https://live-azs-uxua.pantheonsite.io/

The Home Page is completely broken.
/home (was set as homepage, title: UX@UA: A UX Community at the University of Arizona)

On the UAS site, there is a URL redirect. I am not sure if that affected the migration and is involved in the issue.

## Proposed resolution
TBD

### Expected behavior

For migrated sites to retain similar functionality to UAS sites.

##  Screenshots
Screen Shot 2022-01-18 at 3 54 12 PM (3)

### Desktop (please complete the following information):
Problem exists on all OS’s, Browsers and devices.

Problem exists

### PR DESCRIPTION
**Pull request by trackleft**
## Description
Migrate front page variable, and also validate nodes before allowing them to be saved.

## Related Issue
#1231

## How Has This Been Tested?
Locally

## Types of changes
**Arizona Quickstart** (install profile, custom modules, custom theme)
- Patch release changes
   - [x] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
- Minor release changes
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- [ ] Other or unknown

**Drupal core**
- Patch release changes
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major or minor level update
- [ ] Other or unknown

**Drupal contrib projects**
- Patch release changes
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major level update
- [ ] Other or unknown

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by camikazegreen** [comment]
The homepage on the Probo build is still showing as /node, which is the same default that it had before and is not the homepage on kitten (although I can't see what that homepage is set to).

**Comment by trackleft** [comment]
Thanks @camikazegreen I had put the file in the wrong folder.

**Comment by camikazegreen** [comment]
`InvalidArgumentException: Source path node/30 has to start with a slash. in Drupal\path_alias\AliasManager->getAliasByPath() (line 185 of /src/az-quickstart-scaffolding/web/core/modules/path_alias/src/AliasManager.php).`

**Comment by trackleft** [comment]
This works now https://cd4f44b8-bfb4-43db-ab3b-7251abe86aa7--site-migration.probo.build/

File: modules/custom/az_migration/migrations/az_site_frontpage.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+id: az_site_frontpage
+label: Site Front Page
+migration_group: az_migration
+migration_tags:
+  - Quickstart Settings Migration
```
Comment (joeparsons) [inline modules/custom/az_migration/migrations/az_site_frontpage.yml:5]: Are we already using this tag on any of our other migrations?  Personally I think it'd be OK to use `Quickstart Content Migration` on this one, too...
```suggestion
  - Quickstart Content Migration
```

File: modules/custom/az_migration/migrations/az_site_frontpage.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      - az_node_flexible_page
+      - az_node_news
+      - az_node_person
+      - az_node_uaqs_basic_page_to_az_page
+
+  'page/front':
+    plugin: concat
+    source:
+      - constants/slash
+      - '@path'
```
Comment (joeparsons) [inline modules/custom/az_migration/migrations/az_site_frontpage.yml:39]: This seems like a reasonable workaround to me.

File: modules/custom/az_migration/migrations/az_site_frontpage.yml (hunk 1 above)
Comment (trackleft) [reply modules/custom/az_migration/migrations/az_site_frontpage.yml:5]: We aren't

File: modules/custom/az_migration/migrations/az_site_frontpage.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  config_name: system.site
+
+migration_dependencies:
+  optional:
+    - az_node_carousel
+    - az_node_event
+    - az_node_flexible_page
+    - az_node_news
+    - az_node_person
+    - az_node_uaqs_basic_page_to_az_page
```
Comment (trackleft) [inline modules/custom/az_migration/migrations/az_site_frontpage.yml:52]:
```suggestion
    - az_event_categories
    - az_flexible_page_categories
    - az_news_tags
    - az_node_uaqs_basic_page_to_az_page
```

File: modules/custom/az_migration/migrations/az_site_frontpage.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      - constants/slash
+      - '@path'
+
+destination:
+  plugin: config
+  config_name: system.site
+
+migration_dependencies:
+  optional:
+    - az_node_carousel
```
Comment (trackleft) [inline modules/custom/az_migration/migrations/az_site_frontpage.yml:47]:
```suggestion
    - az_event_categories
    - az_flexible_page_categories
    - az_news_tags
    - az_node_carousel
```

**Review by joeparsons — APPROVED** [review]
Looks good to me.

**Review by jwmoore1 — APPROVED** [review]

**Review by akslay — APPROVED** [review]

**Pull request merged**